src/transformStateWithClones.js

```diff
@@ -6,8 +6,40 @@
  *
  * @return {Object[]}
  */
+const ADD_PROPERTIES = 'addProperties';
+const REMOVE_PROPERTIES = 'removeProperties';
+const CLEAR = 'clear';
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  let tempState = { ...state };
+  const stateHistory = [];
+
+  for (const action of actions) {
+    const { type, extraData, keysToRemove } = action;
+
+    switch (type) {
```

Comment:
[CODE STYLE]: It's good practice to include a default case in your switch statement for error handling. This will ensure that any unexpected action types are caught and handled appropriately.